File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string log_line) {
        auto split_at{log_line.find(": ")};
        return log_line.substr(split_at + 2);
    }

    std::string log_level(std::string log_line) {
        auto split_at{log_line.find("]")};
        return log_line.substr(1, split_at - 1);
    }

    std::string reformat(std::string log_line) {
        return message(log_line) + " (" + log_level(log_line) + ")";
    }
} // namespace log_line

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class LogLevels {
    private static final String LEVEL = "\\[(INFO|WARNING|ERROR)\\]:";

    public static String message(String logLine) {
        return logLine.replaceFirst(LEVEL, "").strip();
    }

    public static String logLevel(String logLine) {
        String level = "";
        Matcher matcher = Pattern.compile(LEVEL).matcher(logLine);
        if (matcher.find()) {
            level = matcher.group(1);
        }
        return level.toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}
